packages/contracts/slither.sh

```diff
@@ -0,0 +1,285 @@
+            none)
+                echo "--no-fail-pedantic"
+                ;;
+            *)
+                echo "[!] Unknown fail-on value $FAIL_ON_LEVEL, ignoring" >&2
+                ;;
+        esac
+
+    fi
+}
+
+install_solc()
+{
+    if [[ -z "$SOLCVER" ]]; then
+        echo "[-] SOLCVER was not set; guessing."
+
+        if [[ -f "$TARGET" ]]; then
+            SOLCVER="$(grep --no-filename '^pragma solidity' "$TARGET" | cut -d' ' -f3)"
+        elif [[ -d "$TARGET" ]]; then
+            pushd "$TARGET" >/dev/null
+            SOLCVER="$(grep --no-filename '^pragma solidity' -r --include \*.sol --exclude-dir node_modules --exclude-dir dist | \
+                       cut -d' ' -f3 | sort | uniq -c | sort -n | tail -1 | tr -s ' ' | cut -d' ' -f3)"
+            popd >/dev/null
+        else
+            echo "[-] Target is neither a file nor a directory, assuming it is a path glob"
+            SOLCVER="$( shopt -s globstar; for file in $TARGET; do
+                            grep --no-filename '^pragma solidity' -r "$file" ; \
+                        done | cut -d' ' -f3 | sort | uniq -c | sort -n | tail -1 | tr -s ' ' | cut -d' ' -f3)"
+        fi
+        SOLCVER="$(echo "$SOLCVER" | sed 's/[^0-9\.]//g')"
+
+        if [[ -z "$SOLCVER" ]]; then
+            # Fallback to latest version if the above fails.
+            SOLCVER="$(solc-select install | tail -1)"
+        fi
+
+        echo "[-] Guessed $SOLCVER."
+    fi
+
+    solc-select install "$SOLCVER"
+    solc-select use "$SOLCVER"
+}
+
+install_node()
+{
+    if [[ -z "$NODEVER" ]]; then
+        NODEVER="node"
+        echo "[-] NODEVER was not set, using the latest version."
+    fi
+
+    wget -q -O nvm-install.sh https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.1/install.sh
+    if [ ! "fabc489b39a5e9c999c7cab4d281cdbbcbad10ec2f8b9a7f7144ad701b6bfdc7  nvm-install.sh" = "$(sha256sum nvm-install.sh)" ]; then
+        echo "NVM installer does not match expected checksum! exiting"
+        exit 1
+    fi
+    bash nvm-install.sh
+    rm nvm-install.sh
+
+    # Avoid picking up `.nvmrc` from the repository
+    pushd / >/dev/null
+    . ~/.nvm/nvm.sh
+    nvm install "$NODEVER"
+    popd >/dev/null
+}
+
+install_foundry()
+{
+    if [[ -d "$TARGET" ]] && [[ -f "$TARGET/foundry.toml" ]]; then
+        echo "[-] Foundry target detected, installing foundry nightly"
+
+        wget -q -O foundryup https://raw.githubusercontent.com/foundry-rs/foundry/36a66c857ff148f6ed007cdcd3402077de3595cf/foundryup/foundryup
+        if [ ! "c0c6711dc39deae65bebcc0320fec1265930895a527e18daa643a708218c07ae  foundryup" = "$(sha256sum foundryup)" ]; then
+            echo "Foundry installer does not match expected checksum! exiting"
+            exit 1
+        fi
+
+        export FOUNDRY_DIR="/opt/foundry"
+        export PATH="$FOUNDRY_DIR/bin:$PATH"
+        mkdir -p "$FOUNDRY_DIR/bin" "$FOUNDRY_DIR/share/man/man1"
+        bash foundryup
+        rm foundryup
+    fi
+}
```

Comment:
Do we need to install a node and foundry? We must already have it in CI/CD and locally. And our repo has requirements for specific version of them to work properly.